packages/ts/react-grid/src/autocrud.tsx

```diff
@@ -0,0 +1,58 @@
+import type { AbstractModel, DetachedModelConstructor } from '@hilla/form';
+import type { GridElement } from '@hilla/react-components/Grid.js';
+import { HorizontalLayout } from '@hilla/react-components/HorizontalLayout.js';
+import { createContext, useState, type JSX } from 'react';
+import { ExperimentalAutoForm } from './autoform';
+import { AutoGrid } from './autogrid';
+import type { CrudService } from './crud';
+
+export type AutoCrudProps<TItem> = Readonly<{
+  service: CrudService<TItem>;
+  model: DetachedModelConstructor<AbstractModel<TItem>>;
+}>;
+
+interface AutoCrudContextType<TItem> {
+  item: TItem | undefined;
+}
+
+export const AutoCrudContext = createContext<AutoCrudContextType<any>>({ item: undefined });
+
+export function ExperimentalAutoCrud<TItem>({ service, model }: AutoCrudProps<TItem>): JSX.Element {
+  const [item, setItem] = useState<TItem | undefined>(undefined);
+  const [refreshTrigger, setRefreshTrigger] = useState(0);
+
+  return (
+    <>
+      <HorizontalLayout>
+        <AutoGrid
+          refreshTrigger={refreshTrigger}
+          service={service}
+          model={model}
+          onActiveItemChanged={(e) => {
+            const activeItem = e.detail.value;
+            (e.target as GridElement).selectedItems = activeItem ? [activeItem] : [];
+          }}
+          onSelectedItemsChanged={(e) => {
+            if (e.detail.value.length === 0) {
+              setItem(undefined);
+            } else {
+              const selectedItem = e.detail.value[0];
+              setItem({ ...selectedItem });
+            }
+          }}
+        ></AutoGrid>
+        <ExperimentalAutoForm
+          disabled={!item}
+          service={service}
+          model={model}
+          item={item}
+          onSubmit={({ item: submittedItem }) => {
+            setItem(submittedItem);
+            // Trigger grid data refresh
+            setRefreshTrigger(refreshTrigger + 1);
```

Comment:
Even though it works, it looks hacky :)
Maybe, we could at least hide the logic in a method called `refresh` and call it here.

Reply:
Shouldn't there be a `refresh`/`reload` function defined in `AutoGrid` and exposed for `AutoCrud` to call? However, in the end, that function should change some state in the `AutoGrid` so that the dependent effect is triggered.

Reply:
The next PR for delete introduces a context with a `refreshGrid` method but this would require some more refactoring then to be able to use the context. Can we postpone it a bit so it will be easier? It will probably be necessary to do the refactoring to support customization of the form and/or layout

Reply:
> Shouldn't there be a refresh/reload function defined in AutoGrid and exposed for AutoCrud to call
As we are talking React, a component cannot really publish API...

Reply:
@sissbruecker  had a prototype for a different approach that could be added later also

Reply:
Isn't there a way to avoid reloading the grid data from backend? For example, getting rid of that `refreshTrigger` state and pass the newly added/edited item and adding/updating that one directly into the existing data? Maybe, we don't want to get rid of `refreshTrigger` completely for cases that needs a reload to happen, but in this case it feels like an overhead.

Reply:
Reloading the data is a good first approach. It can always be improved although it requires quite a bit more code. Also by reloading you get changes done by other people. Create an issue and we won't forget it